tests/fracs/test_msh_2_grid.py

```diff
@@ -0,0 +1,308 @@
+"""Test functions in :module:`porepy.fracs.msh_2_grid`.
+
+For now, the tests focus on ensuring that geometries with inclusions are properly
+handled, with the inclusion tag correctly applied to grids of the appropriate
+dimension.
+
+The functions in the module msh_2_grid are also tested indirectly through the tests of
+fracture mesh generation in :mod:`tests.fracs.test_fracture_network_2d` and
+:mod:`tests.fracs.test_fracture_network_3d`.
+
+"""
+
+import copy
+import pathlib
+
+import numpy as np
+import pytest
+
+import gmsh
+
+import porepy as pp
+from porepy.fracs.fracture_network_2d import FractureNetwork2d
+from porepy.fracs.fracture_network_3d import FractureNetwork3d
+from porepy.fracs.msh_2_grid import (
+    create_0d_grids,
+    create_1d_grids,
+    create_2d_grids,
+    create_3d_grids,
+)
+from porepy.fracs.simplex import _read_gmsh_file
+from porepy.grids.mdg_generation import _preprocess_simplex_args
+
+dirname: pathlib.Path = pathlib.Path(__file__).parent
+
+
+INCLUSION_NAME: str = "inclusion"
+
+
+@pytest.fixture(scope="function")
+def dims(request: pytest.FixtureRequest) -> tuple:
+    return request.param
+
+
+@pytest.fixture(scope="function")
+def fracture_network(dims: tuple) -> pp.fracture_network:
+    if len(dims) == 2:
+        bounding_box = {
+            "xmin": 0,
+            "xmax": dims[0],
+            "ymin": 0,
+            "ymax": dims[1],
+        }
+        # Two intersecting line fractures.
+        fractures = pp.fracture_sets.orthogonal_fractures_2d(size=dims[0])
+
+    if len(dims) == 3:
+        bounding_box = {
+            "xmin": 0,
+            "xmax": dims[0],
+            "ymin": 0,
+            "ymax": dims[1],
+            "zmin": 0,
+            "zmax": dims[2],
+        }
+        # Three intersecting plane fractures creating both a line fracture intersection
+        # and a point fracture intersection.
+        fractures = pp.fracture_sets.orthogonal_fractures_3d(size=dims[0])
+    domain = pp.Domain(bounding_box)
+    return pp.create_fracture_network(fractures, domain)
+
+
+@pytest.fixture(scope="function")
+def create_gmsh_file(
```

Comment:
Excellently documented function! Is there any chance it could be handy to reuse it anywhere else? If so, consider moving to applications.test... or something

Reply:
Thanks. For the moment, I would say no, this is highly tailored to the need of this test (and we rarely if ever do this kind of operations).